fluentd/run.sh

```diff
@@ -123,13 +112,58 @@ if [ "${USE_MUX_CLIENT:-}" = "true" ] ; then
     DIV=`expr $DIV \* 2`
 fi
 
+# Default buffer_type: file if FILE_BUFFER_PATH is specified.
+FILE_BUFFER_PATH=${FILE_BUFFER_PATH:-""}
+if [ "$BUFFER_TYPE" = "file" -a "$FILE_BUFFER_PATH" != "" ] ; then
+    if [ ! -d $FILE_BUFFER_PATH ]; then
+        mkdir -p $FILE_BUFFER_PATH
+    fi
+    # Get the available disk size; use 1/4 of it
+    DF_LIMIT=$(df -B1 $FILE_BUFFER_PATH | grep -v Filesystem | awk '{print $2}')
+    DF_LIMIT=${DF_LIMIT:-0}
+    DF_LIMIT=$(expr $DF_LIMIT / 4)
+    # Given buffer limit per output
+    TOTAL_LIMIT=$(echo ${FILE_BUFFER_LIMIT:-0} | sed -e "s/[Kk]/*1024/g;s/[Mm]/*1024*1024/g;s/[Gg]/*1024*1024*1024/g;s/i//g" | bc)
+    # In total
+    TOTAL_LIMIT=$(expr $TOTAL_LIMIT \* $DIV)
+    # Available disk space is less than the specified size.
+    if [ $DF_LIMIT -gt 0 -a $DF_LIMIT -lt $TOTAL_LIMIT ]; then
+        TOTAL_LIMIT=$DF_LIMIT
```

Comment:
Should print a warning here that available disk space $DF_LIMIT is less than requested $TOTAL_LIMIT